src/doc/CMakeLists.txt

```diff
@@ -1,13 +1,19 @@
 find_package ( Doxygen )
 
-file ( MAKE_DIRECTORY ${CMAKE_SOURCE_DIR}/../install/doc)
-
 if (DOXYGEN_FOUND)
-  set (generated_doc_location ${CMAKE_SOURCE_DIR}/../install/doc/ptex/)
-  set (doxyfile_location ${CMAKE_SOURCE_DIR}/doc/)
+  set (generated_doc_location ${CMAKE_CURRENT_BINARY_DIR}/doc)
+  set (doxyfile_location ${CMAKE_CURRENT_SOURCE_DIR} )
   set (perm OWNER_READ GROUP_READ WORLD_READ)
-  add_custom_target (doc
-      ${DOXYGEN_EXECUTABLE} ${doxyfile_location}/Doxyfile
-      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/doc
+  file ( READ ${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile doxygen_configuration_file )
+  file ( WRITE ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile ${doxygen_configuration_file} )
+  file ( APPEND ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile "OUTPUT_DIRECTORY = ${generated_doc_location}\n" )
+  file ( APPEND ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile "INPUT = ${CMAKE_SOURCE_DIR}/ptex ${CMAKE_CURRENT_SOURCE_DIR}\n" )
```

Comment:
This should probably also use `${PROJECT_SOURCE_DIR}` rather than `${CMAKE_SOURCE_DIR}`